x-pack/plugins/ml/public/explorer/explorer_swimlane.js

```diff
@@ -343,109 +349,104 @@ export class ExplorerSwimlane extends React.Component {
     }
 
     const that = this;
-    _.each(lanes, (lane) => {
-      const $lane = $('<div>', {
-        class: 'lane',
-      });
-
-      const label = mlEscape(lane);
-      const fieldName = mlEscape(swimlaneData.fieldName);
-      const laneDivProps = {
-        class: 'lane-label',
-        css: {
-          width: laneLabelWidth + 'px'
-        },
-        html: label
-      };
 
-      if (swimlaneData.fieldName !== undefined) {
-        laneDivProps['tooltip-html-unsafe'] = `${fieldName}: ${label}`;
-        laneDivProps['tooltip-placement'] = 'right';
-        laneDivProps['aria-label'] = `${fieldName}: ${label}`;
-      }
+    const d3Lanes = swimlanes.selectAll('.lane').data(lanes);
+    const d3LanesEnter = d3Lanes.enter().append('div').classed('lane', true);
 
-      const $label = $('<div>', laneDivProps);
-      $label.on('click', () => {
+    d3LanesEnter.append('div')
+      .classed('lane-label', true)
+      .style('width', `${laneLabelWidth}px`)
+      .html(label => mlEscape(label))
+      .on('click', () => {
         if (typeof appState.mlExplorerSwimlane.selectedLanes !== 'undefined') {
           that.clearSelection();
         }
+      })
+      .each(function () {
+        if (swimlaneData.fieldName !== undefined) {
+          d3.select(this)
+            .attr('tooltip-html-unsafe', label => `${mlEscape(swimlaneData.fieldName)}: ${mlEscape(label)}`)
+            .attr('tooltip-placement', 'right')
+            .attr('aria-label', label => `${mlEscape(swimlaneData.fieldName)}: ${mlEscape(label)}`);
+        }
       });
-      $lane.append($label);
 
-      const $cellsContainer = $('<div>', {
-        class: 'cells-container'
-      });
-      $lane.append($cellsContainer);
+    function cellMouseOverFactory(time, i) {
+      // Don't use an arrow function here because we need access to `this`,
+      // which is where d3 supplies a reference to the corresponding DOM element.
+      return function (lane) {
+        const bucketScore = getBucketScore(lane, time);
+        if (bucketScore === 0) { return; }
 
-      const cellMouseOverFactory = (safeLaneTxt, bucketScore, i, time) => {
-        return (e) => {
-          cellMouseover(e, safeLaneTxt, bucketScore, i, time);
-        };
+        const safeLaneTxt = lane.replace(/(['\\])/g, '\\$1');
```

Comment:
Have you tested this with any datasets which contain partitioning field names and values with 'special' characters in, such as quotes, spaces, backslashes. Just worth double checking that all the functionality works as expected.